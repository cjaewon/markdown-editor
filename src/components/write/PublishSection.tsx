import * as React from 'react';
import styled from 'styled-components';
import palette from '../../lib/styles/palette';

const PublishSectionBlock = styled.section`
  & > h3 {
    font-size: 1.3125rem;
    color: ${palette.gray8};
    line-height: 1.5;
    margin-bottom: 0.5rem;
    margin-top: 0;
  }
  .contents {
  }

  & + & {
    margin-top: 1.5rem;
  }
`;

export interface PublishSectionProps extends React.HTMLProps<HTMLElement> {
  title: string;
  children: React.ReactNode;
}

const PublishSection: React.FC<PublishSectionProps> = ({
  title,
  children,
  ...rest
}) => {
  const htmlProps = rest as any;
  return (
    <PublishSectionBlock {...htmlProps}>
      <h3>{title}</h3>
      <div className="contents">{children}</div>
    </PublishSectionBlock>
  );
};

export default PublishSection;
