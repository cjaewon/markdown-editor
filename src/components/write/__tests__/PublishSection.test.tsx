import * as React from 'react';
import { render } from '@testing-library/react';
import PublishSection, { PublishSectionProps } from '../PublishSection';

describe('PublishSection', () => {
  const setup = (props: Partial<PublishSectionProps> = {}) => {
    const initialProps: PublishSectionProps = {
      title: '섹션 제목',
      children: '섹션 내용',
    };
    const utils = render(<PublishSection {...initialProps} {...props} />);
    return {
      ...utils,
    };
  };
  it('renders properly', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('renders title and children', () => {
    const utils = setup();
    utils.getByText('섹션 제목');
    utils.getByText('섹션 내용');
  });
});
