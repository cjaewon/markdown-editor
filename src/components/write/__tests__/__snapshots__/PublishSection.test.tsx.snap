// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishSection matches snapshot 1`] = `
<div>
  <section
    class="sc-bdVaJa ezdyV"
  >
    <h3>
      섹션 제목
    </h3>
    <div
      class="contents"
    >
      섹션 내용
    </div>
  </section>
</div>
`;
